### DISCUSSION TITLE
Chargestorm Connected DUAL outlets

### DISCUSSION THREAD

**Discussion by pretori** [lbbrhzn/ocpp#594] [Q&A]
Hi,

I have a Chargestorm Connected, EV charger with 2 connectors, the integration only seems to have support for one connector.
In the app "Taking Charge" I can controll both outlets.

Has anyone used the intergration with a EV device that has dual connectors?

**Comment by jthun** [comment]
Hi,

I have the Chargestorm Connected with two connectors. The integration sort of works (reports values) but controlling charging doesn't. I disabled the integration so that I could use the manual scheduling in the "Taking Charge" app for now.

It would be really nice to have good support for two connectors, and to be able to control them separately with a dynamic schedule in HA, based on the hourly energy price.

**Reply by ernst77** [reply]
Do you get a separate report for each connector or combined?

**Reply by pretori** [reply]
The unit report each outlet and if you have loadbalancing you can get the that value aswell

**Reply by ernst77** [reply]
are we talking about this ocpp integration or you talking about the app?

**Reply by pretori** [reply]
Sorry, the taking charge app. In that one I can see both. The values are also sent with MQTT to the home assistant server. 

The ctek backend uses occp (taking charge)

**Reply by ernst77** [reply]
I am with the same problem, a different charger. I've made an issue as it looks like a problem, at the moment as well able to see only one connector usage and control only 1 connector.

**Reply by jthun** [reply]
I made some changes to the ocpp integration so that it adds two connectors in addition to the charger and central devices. Seems to work ok. Unzip the attached file into custom_components (instead of the orginial ocpp integration) and re-add it. Provided as-is.

[ocpp.zip](https://github.com/lbbrhzn/ocpp/files/11025943/ocpp.zip)

**Reply by ernst77** [reply]
> I made some changes to the ocpp integration so that it adds two connectors in addition to the charger and central devices. Seems to work ok. Unzip the attached file into custom_components (instead of the orginial ocpp integration) and re-add it. Provided as-is.
> 
> [ocpp.zip](https://github.com/lbbrhzn/ocpp/files/11025943/ocpp.zip)

Amazing, could you do pull request with this repo? this would help all of us.

**Reply by jthun** [reply]
Not sure if its universally a good idea to introduce more devices in the integration, but I can have a look at it.

**Reply by ernst77** [reply]
well, this integration does not support chargers with multiple connectors, so it's not introducing more devices it's actually fixing a bug, people with one connector will keep the functionality, but people who have chargers with multiple connectors will be able to use this integration as well. 
Btw I've forked this repo and introduced your changes in my repo (because your code was 4 months old and did not have some bug fixes). Anyway, BIG THANKS for the code, I am not a phyton/HA developer so this would've taken me a long time.

**Reply by jthun** [reply]
You're welcome! Feel free to use it in a PR against the ocpp repo if you want to.
